airflow/providers/databricks/sensors/databricks_table_changes.py

```diff
@@ -0,0 +1,170 @@
+    :param _schema: An optional initial schema to use.
+        Requires DBR version 9.0+ (templated), defaults to "default"
+    :param table_name: Table name to generate the SQL query, defaults to ""
+    :param handler: Handler for DbApiHook.run() to return results, defaults to fetch_all_handler
+    :param client_parameters: Additional parameters internal to Databricks SQL Connector parameters.
+    :param timestamp: Timestamp to check event history for a Delta table,
+        defaults to datetime.now()-timedelta(days=7)
+    :param filter operator: Operator to specify filter condition to check table changes.
+    """
+
+    template_fields: Sequence[str] = ("databricks_conn_id", "_catalog", "_schema", "table_name", "timestamp")
+
+    def __init__(
+        self,
+        table_name: str,
+        timestamp: datetime = datetime.now() - timedelta(days=7),
+        change_filter_operator: str = ">=",
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.timestamp = timestamp
+        self.caller = "DatabricksTableChangesSensor"
+        self.change_filter_operator = change_filter_operator
+        self.table_name = table_name
+
+    def _get_hook(self) -> DatabricksSqlHook:
+        return DatabricksSqlHook(
+            self.databricks_conn_id,
+            self._http_path,
+            self._sql_endpoint_name,
+            self.session_config,
+            self.http_headers,
+            self._catalog,
+            self._schema,
+            self.caller,
+            **self.client_parameters,
+            **self.hook_params,
+        )
+
+    def _sql_sensor(self, sql):
+        hook = self._get_hook()
+        sql_result = hook.run(
+            sql,
+            handler=self.handler if self.do_xcom_push else None,
+        )
+        return sql_result
+
+    def _generate_query(
+        self,
+        prefix: str,
+        suffix: str,
+        joiner_val: str,
+        table_name: str,
+        time_range: str,
+        operation_filter: str,
+    ) -> str:
+        formatted_opts = ""
+        formatted_opts = f"{prefix} {table_name}{suffix} {joiner_val} '{time_range}'{operation_filter}"
+        self.log.debug("Formatted options: %s", formatted_opts)
+
+        return formatted_opts.strip()
+
+    @staticmethod
+    def get_previous_version(context: Context, lookup_key):
+        return context["ti"].xcom_pull(key=lookup_key, include_prior_dates=True)
+
+    @staticmethod
+    def set_version(context: Context, lookup_key, version):
+        context["ti"].xcom_push(key=lookup_key, value=version)
+
+    def get_current_table_version(self, table_name, time_range, operator):
+        _count_describe_literal = "SELECT MAX(version) AS versions FROM (DESCRIBE HISTORY"
+        _filter_predicate_literal = ") WHERE timestamp"
+        _operation_filter_literal = "AND operation NOT LIKE '%CONVERT%' AND operation NOT LIKE '%OPTIMIZE%' \
```

Comment:
could be simpler to do `operation NOT IN ('CONVERT', 'OPTIMIZE', ...)` ?

Reply:
Also, need to add `FSCK`. See full list here: https://docs.delta.io/latest/delta-utility.html#retrieve-delta-table-history

Reply:
Yes, added `FSCK`. Agree, I wanted to do a simpler `not in`, but the `VACUUM` and a few other commands have a `VACCUM START`, `VACUUM END` etc., so I wanted to capture them all to ensure we had only table data related operations' versions returned.

Reply:
Regarding `VACUUM START`, ... let's try to use just single `%` -> `VACUUM%` to avoid for searching by substring

Reply:
Good point, changed it.